docs/okeps/okep-5085-localnet-api.md

```diff
@@ -0,0 +1,825 @@
+
+#### Suggested API validations
+- `Role`:
+    - Required.
+    - When `topology=Localnet`, the only allowed value is `Secondary`.
+        - Having Role explicitly makes the API predictable and consistent with other topologies. In addition, it enables extending localnet to support future role options
+- `PhysicalNetworkName`:
+    -  Required.
+    - Max length 253.
+    - Cannot contain `,` or `:` characters.
+- `MTU`:
+    - Minimum 576 (minimal for IPv4). Maximum: 65536.
+    - When Subnets consist of IPv6 CIDR, minimum MTU should be 1280.
+- `VLAN`:
+  - `Mode`: Can be "Access".
+  - `ID` (`access.id`)
+    According to [dot1q (IEEE 802.1Q)](https://ieeexplore.ieee.org/document/10004498),
+    VID (VLAN ID) is 12-bits field, providing 4096 values; 0 - 4095. <br/>
+    The VLAN IDs `0`, `1` and `4095` are reserved. <br/>
+    Suggested validations:
+      - Minimum: 2, Maximum: 4094.
+- `Subnets`:
+    - Minimum items 1, Maximum items 2.
+    - Items are valid CIDR (e.g.: "10.128.0.0/16")
+    - When 2 items are specified they must be of different IP families.
+- `ExcludeSubnets`:
+    - Minimum items 1, Maximum items - 25.
+    - Items are valid CIDR (e.g.: "10.128.0.0/16")
+    - Cannot be set when Subnet is unset or `ipam.mode=Disabled`.
+    - Ensure excluded subnet in range of at least on subnet in `spec.network.localnet.subnets`
+      - Due to a bug in Kubernetes CEL validation IP/CIDR operations this validation can be implemented once the following issue is resolved
+        https://github.com/kubernetes/kubernetes/issues/130441 
+        The CRD controller should validate excludeSubnets items are in range of specified subnets. 
+        In a case of an invalid request raise an error in the status.
+
+#### YAML examples
+Assuming the node has OVS bridge-mapping defined by [Kubernetes-nmstate](https://nmstate.io/kubernetes-nmstate/) 
+using the following `NodeNetworkConfigurationPolicy` (NNCP):
+```yaml
+...
+desiredState:
+    ovn:
+      bridge-mappings:
+      - localnet: tenantblue 
+        bridge: br-ex
+```
+Example 1:
+```yaml
+---
+apiVersion: k8s.ovn.org/v1
+kind: ClusterUserDefinedNetwork
+metadata:
+  name: test-net
+spec:
+  namespaceSelector:
+    matchExpressions:
+    - key: kubernetes.io/metadata.name
+      operator: In
+      values: ["red", "blue"]
+  network:
+    topology: Localnet
+    localnet:
```

Comment:
If I understand correctly, this defines a new subtree which uses fields that already appear in other topologies?
I guess that is what "discriminated-unions" means. I do not like these, IMO they hint of a lack of proper modeling.
But I guess if this is already the current pattern used, nothing we can do about it.